libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/useShareDialogBase.ts

```diff
@@ -70,12 +75,37 @@ const useShareDialogState = (
     }, []);
 
     const onGranularGranteeShareChange = useCallback(
-        (grantee: GranteeItem) => {
-            onGranteeAdd(grantee);
+        (grantee: IGranularGranteeUser | IGranularGranteeGroup) => {
+            const originalGrantee = originalGranularGrantees.find((g) => areObjRefsEqual(g.id, grantee.id));
+
+            if (!originalGrantee) {
+                const original = grantees
+                    .filter(isGranularGrantee)
+                    .find((g) => areObjRefsEqual(g.id, grantee.id));
+
+                onGranteeAdd(grantee);
```

Comment:
could you use `setGranteesToAdd` here even though it's a bit longer? It would make the branches of `if` more similar and thus easier to understand